libraries/AP_HAL/examples/RCInput/RCInput.cpp

```diff
@@ -34,7 +34,7 @@ void loop(void)
         }
     }
     if (changed) {
-        for (uint8_t i=0; i<max_channels; i++) {
+        for (uint8_t i=0; i<=max_channels; i++) {
```

Comment:
> With this change we are possibly accessing last_value[MAX_CHANNELS]

I believe that's not true, since `i < nchannels <= MAX_CHANNELS` is an invariant in the previous loop, which means that `max_channels < MAX_CHANNELS`.

I just think `max_channels` is badly named. If it was `max_channel` (as in "maximum channel number", then things would be clearer as well as this pull request.